core-scroll-header-panel.html

```diff
@@ -337,7 +357,35 @@
       this.y = y;
       
       if (event) {
-        this.fire('scroll', {target: this.scroller}, this, false);
+        this.fire('scroll', {target: this.scrollObject}, this, false);
+      }
+    },
+
+    fullWindowSetup: function() {
+      this.scrollObject && this.scrollObject.removeEventListener('scroll', this._scrollHandler);
+      this.scrollObject = this.fullWindow ? window : this.$.mainContainer;
+      this.scrollObject.addEventListener('scroll', this._scrollHandler);
+      if (this.fullWindow) {
+        this.style.position = 'static';
+        this.style.height = 'auto';
+        if (!this.drawerPanel) {
```

Comment:
... you are right, the easy way might not necessarily be the best. Assigning an element to a local variable allows binding to all its public attributes and methods. Yet I agree, it does not look (and feel) good.